README.md

```diff
@@ -2012,8 +2012,67 @@ final class InvalidTest extends TestCase
 }
 ```
 
-| :white_check_mark: **GOOD:** |
-|:-----------------------------|
+| :exclamation: **WARNING:** |
+|:---------------------------|
+Sharing an object state during `setUp` couples test methods together.
+
+| :white_check_mark: **BETTER:** |
+|:-------------------------------|
+
+```php
+final class BetterTest extends TestCase
+{
+    private Subscription $sut;
+
+    public function setUp(): void
```

Comment:
Hmm, I think that this example can be a bit tricky, maybe there should be two separate examples:
- one with something when using the setUp method is ok, because that object is the same in all cases e.g. some kind of stateless object
- second one without using setUp, but using private methods

Then I think we can achieve the best clarity of those examples.  What do you think?

Reply:
AD1) What do you mean? According to phpunit's docs using, setup" _The setUp() and tearDown() template methods are run once for each test method (and on fresh instances) of the test case class._"

AD2) I'm not sure about the second point because to me this would look the same as `ValidTest`. 
Maybe in context of this, I should add `BEST` mark. I think I missed that.

Reply:
> AD1) What do you mean? According to phpunit's docs using, setup" The setUp() and tearDown() template methods are run once for each test method (and on fresh instances) of the test case class."

I mean that the setUp method should be used only for common things for all tests, because this method introduces coupling between all tests. The best case for this will be testing stateless objects. But still, the readability is worse compared to creating all necessary data in the proper test method.

> AD2) I'm not sure about the second point because to me this would look the same as ValidTest.
Maybe in context of this, I should add BEST mark. I think I missed that.

I mean that we should have only two examples for Test fixtures section as I described above. This example with __construct is a rather trivial one, I didn't see that mistake in projects.

Reply:
👍🏻  ok, I think I got it now 😅 , so just to quickly recap:

1) In current context to me, if we use `setUp` in `InvalidTest` – that test would become let's say `BasicTest`, and not an invalid one, but some explanation about worse readability could be added
2) `ValidTest` would become `BetterTest` where setup could be replaced with only private methods or some testing patterns.

Reply:
Yeah, exactly ;) But we have to think about a class to test to represent `GoodTest` example. It should be rather something with pure logic without any state and then we'll have only input and output and using the setUp method will be useful. Maybe some kind of domain service?